src/libsyncengine/jobs/abstractjob.h

```diff
@@ -115,13 +118,20 @@ class AbstractJob : public Poco::Runnable {
 
         std::function<void(UniqueId)> _mainCallback = nullptr;        // Used by the job manager to keep track of running jobs
         std::function<void(UniqueId)> _additionalCallback = nullptr;  // Used by the caller to be notified of job completion
+        std::function<void(UniqueId id, int progress)> _progressPercentCallback =
+            nullptr;  // Used by the caller to be notified of job progress change
 
         static UniqueId _nextJobId;
         static std::mutex _nextJobIdMutex;
 
         UniqueId _jobId = 0;
         UniqueId _parentJobId = -1;  // ID of that parent job i.e. the job that must be completed before starting this one
 
+        int64_t _expectedFinishProgress = -1;  // Expected progress value when the job is finished
+        int64_t _progress = -1;      // Progress is -1 when it is not relevant for the current job
+        int64_t _lastProgress = -1;  // Progress last time it was checked using progressChanged()
+        SyncPath _affectedFilePath;  // The file path associated to _progress
```

Comment:
This variable name does relate to the idea of progress. 
Question: Is a measure progress in the context of `AbstractJob` always related to a file? 
What is the rôle of this file?

Reply:
It is, in fact, used by SyncPal to set the status of the file. It may be redundant with _localPath, but it usually stores a relative path, whereas _localPath uses an absolute path.